src/RequestHeaderParser.php

```diff
@@ -85,13 +87,34 @@ private function parseRequest($data)
 
         // create new obj implementing ServerRequestInterface by preserving all
         // previous properties and restoring original request-target
+        $serverParams = array(
+            'REQUEST_TIME' => time(),
+            'REQUEST_TIME_FLOAT' => microtime(true)
+        );
+
+        if (!empty($this->remoteAddress)) {
```

Comment:
Not a big fan of `empty()` here (`"0"` could technically be a valid UDS address). Does it make sense to use a `null` check here and below?